### PR TITLE
servo_write now automatically sends header and checksum

### PR DESCRIPTION
**Pull request by chqoot-vgxu**

### CODE REVIEW

File: nightmare_firmware/servos.cpp [hunk 1]
```diff
@@ -65,13 +71,11 @@ void servo::move(int ang){
...
-		SERVO_MOVE_TIME_WRITE, 
-		ang & 0xFF, 
-		ang >> 8, 
-		0x00, 0x00, 
-		~((byte)(id+0x08+(ang & 0xFF)+(ang >> 8)))
+		id,
+		SERVO_MOVE_TIME_WRITE,
+		ang & 0xFF,
+		ang >> 8,
+		0x00, 0x00
```
Comment (Cogno-Marco) [inline nightmare_firmware/servos.cpp:78]: Are you sure it stays the same?
checksum is everything passed into servo_write and added together, then inverted (using `~`)
old code the checksum was `~((byte)(id+0x08+(ang & 0xFF)+(ang >> 8)))`, while new code checksum is `~(id + SERVO_MOVE_TIME_WRITE + (and & 0xFF) + (ang>>8))` right? Aren't you missing `id + 0x08`?

File: nightmare_firmware/servos.cpp [hunk 2]
```diff
@@ -100,11 +102,9 @@ void servo::attach(){
   	servo_tx_enb();
   
   	servo_write(
-		SERVO_HEADER,
 		id,
 		SERVO_LOAD_OR_UNLOAD_WRITE,
-		0x01,
-		~((byte)(id+0x24))
+		0x01
   	);
```
Comment (Cogno-Marco) [inline nightmare_firmware/servos.cpp:108]: same here, aren't you missing something?here you get `id+0x01+4+31` (4 and 31 are from `SERVO_LOAD_OR_UNLOAD_WRITE`)

File: nightmare_firmware/servos.cpp [hunk 3]
```diff
@@ -135,12 +133,7 @@ int servo::read(){
   	Serial2.flush();                 //wait for bus to clear
   	servo_tx_enb();                  //enable write mode
 
-  	servo_write(                     //request angle
-		SERVO_HEADER,
-		id,
-		SERVO_POS_READ,
-		~((byte)(id+0x1F))
-  	);
+  	servo_write(id, SERVO_POS_READ); //request angle
```
Comment (Cogno-Marco) [inline nightmare_firmware/servos.cpp:136]: And here too, from `id+0x1F` to `id + SERVO_POS_READ` = `id+3+28`

**Review by Cogno-Marco — CHANGES_REQUESTED** [review]
Overrall a nice change, but are you sure it is working?

File: nightmare_firmware/servos.cpp (hunk 1 above)
Comment (Cogno-Marco) [reply nightmare_firmware/servos.cpp:78]: EDIT: Oooh now I see, SERVO_MOVE_TIME_WRITE "=" 1 + 7 so you get that 0x08 in the end, same for others, alright, marking this as resolved

**Review by Cogno-Marco — APPROVED** [review]
Alright, I'm stupid, now I get how it works, seems fine to me then, really cool change 👍

**Review by nikisalli — APPROVED** [review]
looks good

**Pull request merged**